Update readme

jmontleon opened the pull request:
**For each of the following check the box when you have verified either:**
* **the changes have been made for each applicable version**
* **no changes are required for the item**
* **PR's that are submitted without running through the list below will be CLOSED**

Affected versions:
* [ ] Latest
* [ ] 1.1
* [ ] 1.0

The CSV is responsible in OLM installs for
* [X] Operator permissions
* [X] Operator deployment
* [X] Operand permissions
* [X] CRDs

The operator.yml is responsible in non-OLM installs for
* [X] Operator permissions
* [X] Operator deployment

The ansible role is responsible in non-OLM installs for:
* [X] Operand permissions
* [X] CRDs

The ansible role is always responsible for:
* [X] Operand deployment

If this PR updates a release or replaces channel 
* [X] I created a new z release directory in `deploy/olm-catalog/konveyor-operator`
* [X] I updated channels in the `konveyor-operator.package.yaml`
* [X] I created a new release directory in `deploy/non-olm`
* [X] I created or updated the major.minor link in `deploy/non-olm`
* [X] Updated the `.github/pull_request_template.md` Affected versions list

djwhatle commented on roles/migrationcontroller/templates/velero_supporting.yml.j2:
```diff
@@ -15,7 +15,7 @@ spec:
   preserveUnknownFields: false
   scope: Namespaced
-{% if lookup('k8s', cluster_info='version').kubernetes.minor|replace('+', '')|int < 11 %}
+{% if lookup('k8s', cluster_info='version').kubernetes.minor|replace('+', '')|int > 10 %}
```
Is this a purposeful change or got pulled in by mistake?

jmontleon replied:
Pulled by mistake. Should only be the readme. Probably need to rebase.

djwhatle commented on README.md:
```diff
@@ -1,62 +1,43 @@
+Konveyor Operator is available in Operator Hub. This is the recommended installation method for OpenShift 4.
+1. Browse or search for Konveyor Operator.
+1. Install the desired version.
+1. For pre-release versions see the development guide above.
```
+1

djwhatle replied:
@jmontleon can we can the text "development guide" a link?

djwhatle commented on README.md:
```diff
@@ -1,62 +1,54 @@
+Component installation and configuration is accomplished by creating or modifying a MigrationController CR.
 
-Edit `controller-3.yml` or `controller-4.yml` and adjust options if desired.
+### Installing Components
```
This bulleted list is very dense with info, a bit hard to parse IMO. 

Maybe it would make sense to split it out to a few sub sections, and maybe use paragraphs instead of bullets if we need to explain a lot of stuff.

**Possible sub-sections to split out for easier reading**
 - Versioned YAML manifests (discuss where to get YAML manifests for latest, 1.1, 1.2)
 - Recommended Install Topologies (discuss where components get installed)
 - Choosing components to install
 - ...

I'm not sure if this layout will fit perfectly with the content but think we could break this section out and benefit in readability gain.

jmontleon replied:
Does what is there now look/read any better/easier?

djwhatle commented on README.md:
```diff
@@ -117,15 +116,19 @@ If you haven't already, determine the CORS URL that needs to be added retrieve t
 `(?i)//migration-openshift-migration\.apps\.foo\.bar\.baz\.com(:|\z)`
 
-#### For Openshift 4.2
+### For Openshift 4.2  Manual Configuration
```
Suggest removing `For`

djwhatle commented on README.md:
````diff
@@ -117,15 +116,19 @@ If you haven't already, determine the CORS URL that needs to be added retrieve t
 ```
 
-#### For OpenShift 4.1:
+### For OpenShift 4.1 Manual Configuration
````
Suggest removing `For`

djwhatle commented on README.md:
```diff
@@ -107,8 +107,7 @@ $ /usr/local/bin/master-restart api
 
-
-### Openshift 4
+### Openshift 4.3+ Manual Configuration
```
All of the manual configuration could go under a larger heading. 

Also, if our operator does this, I'd argue we don't need to include the manual steps on our front-page README. User only needs to know about it if they're going to have to tweak it.

---

## Manual CORS Configuration

Manual CORs config is needed on these versions:
 - OpenShift x.x
 - OpenShift x.y

### OpenShift 4.2
[... config steps here ...]

### OpenShift 4.3
[... config steps here ...]

### OpenShift 4.4
[... config steps here ...]

jmontleon replied:
I made them sub sections like you suggested. I considered removing the section altogether or putting it in another file. I'm a little torn on this since we have people expressing interest in 3 -> 3 migrations and it will apparently remain relevant in current versions after talking to @eriknelson.

jmontleon replied:
What do you guys think is best?

djwhatle replied:
My vote would be moving CORs config info to another file and linking to that file from here.

jmontleon commented:
@djwhatle Thanks for taking a look. I fixed some typos, added some additional instruction in the installation section, etc. I would appreciate it if you had another minute to look.

I also removed the controller-4.yml's. I think I want to strat discouraging their use, but haven't deleted them in case developers want to use them a bit longer until they've had time to change their workflows. In general I feel OLM is easy enough to use, even if looking for a pre-release version that we should be focusing on that installation method.
